episode-11/main.py

```diff
@@ -58,22 +76,12 @@ def update(self, delta_time):
 	def on_draw(self):
 		self.camera.update_matrices()
 
-		# bind textures
-
-		gl.glActiveTexture(gl.GL_TEXTURE0)
-		gl.glBindTexture(gl.GL_TEXTURE_2D_ARRAY, self.world.texture_manager.texture_array)
-		gl.glUniform1i(self.shader_sampler_location, 0)
-
-		# draw stuff
-
-		gl.glEnable(gl.GL_DEPTH_TEST)
-		gl.glEnable(gl.GL_CULL_FACE)
-
 		gl.glClearColor(0.0, 0.0, 0.0, 0.0)
 		self.clear()
-		self.world.draw()
 
-		gl.glFinish()
+		self.world.draw()
+		
+		gl.glClientWaitSync(self.fence, gl.GL_SYNC_FLUSH_COMMANDS_BIT, 16666667)
```

Comment:
Were you ever able to reproduce #10 and is there a chance that this fixes it? Not in a position to try this out myself...

Reply:
I dont really have #10 and this is just an attempt to use sync objects, rn I just pass an arbitrary value, it should behave similar to glFinish()

Reply:
What GPU again?